apps/src/templates/instructions/DocumentationTab.jsx

```diff
@@ -1,13 +1,47 @@
-import React from 'react';
+import React, {useEffect, useState} from 'react';
 import {connect} from 'react-redux';
 import PropTypes from 'prop-types';
+import Spinner from '@cdo/apps/code-studio/pd/components/spinner';
 
 /**
  * This is a placeholder for the upcoming "Documentation" tab in Javalab.
  * See related task: https://codedotorg.atlassian.net/browse/CSA-361
  */
 const UnconnectedDocumentationTab = function({programmingEnvironment}) {
-  return <p>Documentation for {programmingEnvironment}</p>;
+  const [documentation, setDocumentation] = useState(null);
+  const [isLoading, setIsLoading] = useState(true);
+  const [loadError, setLoadError] = useState(false);
```

Comment:
If it helps, you might be able to use [LoadableComponent.jsx](https://github.com/code-dot-org/code-dot-org/blob/staging/apps/src/templates/LoadableComponent.jsx) to abstract away some of this boilerplate

Reply:
oh this is cool, will try it out!

Reply:
Just to throw it out there, another similar option is [useFetch](https://github.com/code-dot-org/code-dot-org/blob/c89d4fad222b64a9a2ac17735377479e7d734fee/apps/src/util/useFetch.js#L38) which I've been (slowly) working on as part of changes to LessonLockDialog.

Reply:
Thanks for the suggestions! I ended up using `useFetch` because `LoadableComponent` didn't play nicely with a result that is an array (because it uses the spread operator).

Reply:
This looks so clean with `useFetch`!